### PR TITLE
Improve `new DefaultChannelId`

### PR DESCRIPTION
**Pull request by jchrys**
Motivation:
The instantiation process of DefaultChannelId presents opportunities for optimization.

Modifications:
Use the local stack when creating an instance.
Employ `Unsafe`.

Result:
Enhanced Performance.

### CODE REVIEW

File: transport/src/main/java/io/netty/channel/DefaultChannelId.java [hunk 1]
```diff
@@ -189,48 +189,59 @@ static int defaultProcessId() {
...
 
-    private int writeInt(int i, int value) {
-        data[i ++] = (byte) (value >>> 24);
-        data[i ++] = (byte) (value >>> 16);
-        data[i ++] = (byte) (value >>> 8);
-        data[i ++] = (byte) value;
-        return i;
+    private static void writeInt(byte[] data, int i, int value) {
+        if (PlatformDependent.isUnaligned()) {
+            PlatformDependent.putInt(data, i, value);
```
Comment (franz1981) [inline transport/src/main/java/io/netty/channel/DefaultChannelId.java:223]: how about the endianess? I don't remember by heart if putInt is BE, if it is, it's ok

Comment (jchrys) [reply transport/src/main/java/io/netty/channel/DefaultChannelId.java:223]: oh, let me fix it.

Comment (jchrys) [reply transport/src/main/java/io/netty/channel/DefaultChannelId.java]: I changed it.

**Comment by jchrys** [comment]
1X10X2, Intel(R) Xeon(R) Silver 4210 CPU @ 2.20GHz, openjdk 17.0.8 2023-07-18, Ubuntu 22.04.3 LTS, tuend network low-latency, no turbo boost.

[latest benchmark result](https://jmh.morethan.io/?sources=https://gist.githubusercontent.com/jchrys/618caaf0696443eb32828a9039789070/raw/6b461b5658a853afd83d7906f46d2b8b38475635/dcidOld.json,https://gist.githubusercontent.com/jchrys/618caaf0696443eb32828a9039789070/raw/6b461b5658a853afd83d7906f46d2b8b38475635/dcidNew.json)

**Review by chrisvest — APPROVED** [review]
In Netty 5 I suspect we'll be better off to just assign instance fields (and bump the `serialVersionUID`) instead of encoding a byte array.

**Comment by jchrys** [comment]
> In Netty 5 I suspect we'll be better off to just assign instance fields (and bump the `serialVersionUID`) instead of encoding a byte array.

I agree. That approach would be straight forward and could save resources by avoiding the assignment & calculation of constant values such as `MACINE_ID` and `PROCESS_ID` as well.

**Comment by chrisvest** [comment]
> That approach would be straightforward and could save resources by avoiding the assignment & calculation of constant values such as `MACINE_ID` and `PROCESS_ID` as well.

There still needs to be instance fields for those, because `ChannelId` is `Serializable`, but they're just plain field assignments.

**Comment by jchrys** [comment]
> There still needs to be instance fields for those, because ChannelId is Serializable, but they're just plain field assignments.

Thank you for clarifying that for me :D

**Comment by jchrys** [comment]
@chrisvest
May I submit a PR for Netty5 based on the information above?

**Comment by chrisvest** [comment]
@jchrys Yes, please go ahead.

**Pull request merged**